Not compatible with python 3.11

```
  gcc -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wno-unused -g -O2 -g -O2 -I/Users/mkoeppe/s/sage/sage-rebasing/worktree-py311/local/var/lib/sage/venv-python3.11.0b1/include -I/Users/mkoeppe/s/sage/sage-rebasing/worktree-py311/local/var/lib/sage/venv-python3.11.0b1/lib/python3.11/site-packages/numpy/core/include -I/Users/mkoeppe/s/sage/sage-rebasing/worktree-py311/local/var/lib/sage/venv-python3.11.0b1/include/python3.11 -c build/src/fpylll/gmp/pylong.c -o build/temp.macosx-12.4-x86_64-3.11/build/src/fpylll/gmp/pylong.o
  build/src/fpylll/gmp/pylong.c:719:10: fatal error: 'longintrepr.h' file not found
  #include "longintrepr.h"
           ^~~~~~~~~~~~~~~
  1 error generated.
  error: command '/usr/bin/gcc' failed with exit code 1
```
https://trac.sagemath.org/ticket/33842

--- a/src/fpylll/gmp/pylong.pxd
+++ b/src/fpylll/gmp/pylong.pxd
@@ -3,7 +3,7 @@
 Various functions to deal with conversion mpz <-> Python int/long
 """
 
-cdef extern from "longintrepr.h":
+cdef extern from "Python.h":
     cdef _PyLong_New(Py_ssize_t s)
     cdef long PyLong_SHIFT
     ctypedef unsigned int digit

